agent/stats/service_connect_linux.go

```diff
@@ -50,24 +54,18 @@ func newServiceConnectStats() (*ServiceConnectStats, error) {
 
 // TODO [SC]: Add retries on failure to retrieve service connect stats
 func (sc *ServiceConnectStats) retrieveServiceConnectStats(task *apitask.Task) {
-	var ipAddress string
-	if task.IsNetworkModeAWSVPC() {
-		ipAddress = task.GetLocalIPAddress()
-	} else {
-		for _, container := range task.Containers {
-			// TODO [SC]: Implement a proper check for appnet agent container when it is defined
-			if strings.Contains(container.Name, "envoy") &&
-				(container.GetNetworkModeFromHostConfig() == "" || container.GetNetworkModeFromHostConfig() == bridgeNetworkMode) {
-				network := *container.GetNetworkSettings()
-				networkSettings := network.DefaultNetworkSettings
-				ipAddress = networkSettings.IPAddress
-				break
-			}
-		}
+
+	serviceConnectConfig := task.GetServiceConnectRuntimeConfig()
+
+	httpclient := http.Client{
```

Comment:
nit: can you add a `TODO [SC]` here to indicate this needs to use the (future) appnet client?

Reply:
sure. Added